frontend/taipy-gui/src/utils/formatConversion.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+/**
+ * Regular expressions used for parsing sprintf format strings.
+ */
+const re = {
+    text: /^[^\x25]+/,                         // Matches non-placeholder text
+    modulo: /^\x25{2}/,                        // Matches the '%%' escape sequence
+    placeholder: /^\x25?(?:\.(\d+))?([b-giostuvxX])/, // Matches placeholders
+};
+
+/**
+ * Parses a sprintf format string and returns a parse tree.
+ * @param fmt The sprintf format string.
+ * @returns The parse tree representing the structure of the format string.
+ */
+const sprintf_parse = (fmt: string) => {
```

Comment:
use camelCase in ts: sprintfParse
`fmt: string` => `fmt?: string`

Reply:
resolved